File: StaLang_Rust/src/ast/tokens.rs

enum TokenKind {
  KEYWORD,
  OPERATOR,
  SEPARATOR
}

struct TokenDocs {
  token_id: i8,
  kind: TokenKind,
  name: String,
  sign: String
}

pub struct TokenOperational {
  sign: String,
  pos_s: i8,
  pos_f: i8,
  line: i8
}

File: ast/tokens.swift

public enum TokenKind {
    case KEYWORD
    case NUMBER
    case STRING
    case SEP
    case OPER
}

public struct TokenDocs {
    var tokenId: Int
    var kind: TokenKind
    var name: String
    var sign: String
    var docs: String
}


/// Token type for opererational goals
public struct TokenInner {
    var tokenId: Int
    var kind: TokenKind
    var sign: String
    var posS: Int
    var posF: Int
    var line: Int
}
